presto-main/src/main/java/io/prestosql/operator/scalar/ArrayContainsSubarray.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.operator.scalar;
+
+import io.prestosql.spi.block.Block;
+import io.prestosql.spi.function.Convention;
+import io.prestosql.spi.function.Description;
+import io.prestosql.spi.function.OperatorDependency;
+import io.prestosql.spi.function.ScalarFunction;
+import io.prestosql.spi.function.SqlNullable;
+import io.prestosql.spi.function.SqlType;
+import io.prestosql.spi.function.TypeParameter;
+import io.prestosql.spi.type.StandardTypes;
+import io.prestosql.spi.type.Type;
+
+import java.lang.invoke.MethodHandle;
+
+import static io.prestosql.spi.function.InvocationConvention.InvocationArgumentConvention.BLOCK_POSITION;
+import static io.prestosql.spi.function.InvocationConvention.InvocationReturnConvention.NULLABLE_RETURN;
+import static io.prestosql.spi.function.OperatorType.IS_DISTINCT_FROM;
+
+@Description("Determines whether an array contains another, with the values in the exact order")
+@ScalarFunction("contains_subarray")
+public final class ArrayContainsSubarray
+{
+    private ArrayContainsSubarray() {}
+
+    @TypeParameter("T")
+    @SqlType(StandardTypes.BOOLEAN)
+    @SqlNullable
+    public static Boolean containsSubArray(
+            @TypeParameter("T") Type elementType,
+            @OperatorDependency(
+                    operator = IS_DISTINCT_FROM,
+                    argumentTypes = {"T", "T"},
+                    convention = @Convention(arguments = {BLOCK_POSITION, BLOCK_POSITION}, result = NULLABLE_RETURN))
+                    MethodHandle distinctFrom,
+            @SqlType("array(T)") Block arrayBlock,
+            @SqlType("array(T)") Block value)
+            throws Throwable
+    {
+        int arrayLimit = arrayBlock.getPositionCount() - value.getPositionCount();
+        for (int arrayPosition = 0; arrayPosition <= arrayLimit; arrayPosition++) {
+            if (containsSubArrayAt(distinctFrom, arrayBlock, arrayPosition, value)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static boolean containsSubArrayAt(MethodHandle distinctFrom, Block arrayBlock, int arrayPosition, Block value)
+            throws Throwable
+    {
+        for (int valuePosition = 0; valuePosition < value.getPositionCount(); valuePosition++) {
+            if ((Boolean) distinctFrom.invokeExact(arrayBlock, arrayPosition + valuePosition, value, valuePosition)) {
```

Comment:
Once result is changed to `FAIL_ON_NULL`, this should change to a primitive `boolean`

Reply:
All done, but some product tests failed, seems not my fault